FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift

```diff
@@ -162,15 +158,48 @@ class EnterpriseServerApi: EnterpriseServerApiType {
         return try await googleService.getCachedOrRefreshedIdToken()
     }
 
-    private func shouldTolerateWhenCallingOpportunistically(_ error: Error) -> Bool {
-        guard
-            let apiError = error as? ApiError,
-            let nsError = apiError.internalError as NSError?,
-            Constants.getToleratedNSErrorCodes.contains(nsError.code)
-        else {
+    private func isExpectedFesServiceResponse(responseData: Data) -> Bool {
+        // "try?" because unsure what server is running there, want to test without failing
+        guard let responseDictionary = try? responseData.toDict() else { return false }
+        guard let service = responseDictionary["service"] as? String else { return false }
+        return service == "enterprise-server"
+    }
+    
+    private func shouldTolerateWhenCallingOpportunistically(_ error: Error) async -> Bool {
+        if Bundle.isEnterprise() {
+            return false // FlowCrypt Enterprise Server (FES) required on enterprise bundle
+        }
+        // on consumer release, FES is called opportunistically - if it's there, it will be used
+        // guards first - don't tolerate unknown / other errors. Only interested in network errors.
+        guard let apiError = error as? ApiError else { return false }
+        guard let nsError = apiError.internalError as NSError? else { return false }
+        guard Constants.getToleratedNSErrorCodes.contains(nsError.code) else { return false }
```

Comment:
For removing duplicated code and making it a bit more readable we can use single `guard` like this:
```swift
guard let apiError = error as? ApiError,
      let nsError = apiError.internalError as NSError?,
      Constants.getToleratedNSErrorCodes.contains(nsError.code)
else { return false }
```

Reply:
I know. I just seem to prefer the readability of separate guards. Since you find a combined guard more readable, I'll combine them back.

Reply:
Yes, it looks a bit easier to read when you have just conditions on each line, without `guard` (I think most Swift style guides use this format too)